code/ATMOSPHERICS/components/unary_devices/outlet_injector.dm

```diff
@@ -168,6 +168,61 @@
 			on = 0
 		return*/
 
+/obj/machinery/atmospherics/unary/outlet_injector/attack_hand(mob/user)
+	if(..())
+		return
+
+	if(!allowed(user))
+		to_chat(user, "<span class='alert'>Access denied.</span>")
+		return
+
+	add_fingerprint(user)
+	ui_interact(user)
+
+/obj/machinery/atmospherics/unary/outlet_injector/attack_ghost(mob/user)
+	ui_interact(user)
```

Comment:
Have you checked that ghosts can't mess with valves? (I would assume they won't be able to, but just making sure)

Reply:
I checked. Ghosts can't mess with them. Ghosts also can't interact with the new element of the computer UI.